genesyscloud/architect_schedules/data_source_genesyscloud_architect_schedules.go

```diff
@@ -0,0 +1,35 @@
+package architect_schedules
+
+import (
+	"context"
+	"fmt"
+	"terraform-provider-genesyscloud/genesyscloud/provider"
+	"terraform-provider-genesyscloud/genesyscloud/util"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceArchitectSchedulesRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	sdkConfig := m.(*provider.ProviderMeta).ClientConfig
+	proxy := newArchitectSchedulesProxy(sdkConfig)
+
+	name := d.Get("name").(string)
+
+	return util.WithRetries(ctx, 15*time.Second, func() *retry.RetryError {
+		scheduleId, retryable, proxyResponse, err := proxy.getArchitectSchedulesIdByName(ctx, name)
+
+		if err != nil && !retryable {
+			return retry.NonRetryableError(fmt.Errorf("error requesting schedule %s: %s %v", name, err, proxyResponse))
```

Comment:
We have added a new wrapper function to use when returning a `retry.RetryError`, it's called `util.BuildWithRetriesApiDiagnosticError`. You'll need to add it here and anywhere we return `retry.RetryError`. The usage is like so
```go
return retry.NonRetryableError(util.BuildWithRetriesApiDiagnosticError(resourceName, fmt.Sprintf("Error grammar %s | error: %s", name, err), resp))
```
This was probably added after you forked the repo so if you rebase with main you should see it in the other resources

Reply:
> We have added a new wrapper function to use when returning a `retry.RetryError`, it's called `util.BuildWithRetriesApiDiagnosticError`. You'll need to add it here and anywhere we return `retry.RetryError`. The usage is like so
> 
> ```go
> return retry.NonRetryableError(util.BuildWithRetriesApiDiagnosticError(resourceName, fmt.Sprintf("Error grammar %s | error: %s", name, err), resp))
> ```
> 
> This was probably added after you forked the repo so if you rebase with main you should see it in the other resources

Sure, will use the wrapper and update the code.

Reply:
As some background on this @sureshperiyappan we are standardizing on a JSON structure for many of these error calls and then making sure they get written out consistently.  Eventually, I want to write some analysis tools to help us more quickly isolate errors